cli/command/context/create.go

```diff
@@ -0,0 +1,105 @@
+package context
+
+import (
+	"github.com/docker/cli/cli"
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/context/docker"
+	"github.com/docker/cli/cli/context/kubernetes"
+	"github.com/docker/cli/cli/context/store"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+func newCreateCommand(dockerCli command.Cli) *cobra.Command {
+	opts := &createOptions{}
+	cmd := &cobra.Command{
+		Use:   "create [OPTIONS] NAME",
+		Short: "Create a context",
+		Args:  cli.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.name = args[0]
+			return runCreate(dockerCli, opts)
+		},
+	}
+	flags := cmd.Flags()
+	flags.StringVar(&opts.description, "description", "", "Description of the context")
+	flags.StringVar(
+		&opts.defaultStackOrchestrator,
+		"default-stack-orchestrator", "",
+		"Default orchestrator for stack operations to use with this context (swarm|kubernetes|all)")
+	flags.BoolVar(&opts.use, "use", false, "Use this context by default")
+	opts.docker.addFlags(flags, "docker-")
+	opts.kubernetes.addFlags(flags, "kubernetes-")
+	return cmd
+}
+
+type createOptions struct {
+	name                     string
+	description              string
+	defaultStackOrchestrator string
+	docker                   dockerEndpointOptions
+	kubernetes               kubernetesEndpointOptions
+	use                      bool
+}
+
+func runCreate(cli command.Cli, o *createOptions) error {
+	if o.name == "" {
```

Comment:
Can `name` be anything? No restriction on it? I see in this current implementation it ends as a directory name, but you mentioned maybe other implementations. I'm just wondering if it's shooting ourselves in the foot or not 🤔

Reply:
It will must be a valid directory name. I let the OS raise an error on directory creation instead of trying to enforce that with some difficult to maintain OS-specific code.